### PR TITLE
569 large facet dropdown

### LINKED ISSUE #569: Facets: Make long facets more usable
- It's clear how the world will be different when the work is completed

We'll have multi-select dropdowns for funding org, recipient org (+ any others that make sense) 

Blank search will show a few examples, so that we still get the benefit of showing people what they can look for

- It's clear what will and won't be done

We'll make multi-select dropdowns work, but styling can be done next sprint as this may require input from a graphic designer

- Any uncertainty, assumptions, dependencies or constraints are recorded

#571 probably needs to be done first, although doing this first might just result in needing to do some rework

- There's an order-of-magnitude estimate of how long it will take

hours per facet

- The work's been approved by someone at ThreeSixtyGiving

Yes 

- There are links to everything that's written down about the work - such as plan.io tickets, GitHub issues

### PR DESCRIPTION
**Pull request by kindly**
Aims to fix #569 and #570

### CODE REVIEW

**Review by michaelwood — APPROVED** [review]

**Comment by michaelwood** [comment]
There is a bit of strange behaviour I found if you use the backspace after selecting an item in the list, it converts the 'selected pill badge' to text, possibly this is a select2 bug?
Screenshot from 2020-11-11 18-26-47

I'm also seeing escaped values in the labels, though I see this also happens elsewhere in GN too. 

Screenshot from 2020-11-11 18-12-00

File: grantnav/frontend/templates/search.html [hunk 1]
```diff
@@ -400,6 +397,81 @@ <h1>Search input is not valid</h1>
     return false
 
   })
+
+  $.fn.select2.amd.require([
+    'select2/utils',
+    'select2/dropdown',
+    'select2/dropdown/attachBody',
+  ], function (Utils, Dropdown, AttachBody) {
+    function Submit() { }
```
[the remark this answers is not in the document]
Comment (kindly) [reply grantnav/frontend/templates/search.html:445]: This is used as a prototype.  The next lines adds  a render method to it.  Select2 needs a prototype with a render method for you to customise the dropdown.
In modern javascript I think a class would work here instead.  I just did it this way as it was how the docs said to do it.

File: grantnav/frontend/templates/search.html [hunk 2]
```diff
@@ -447,29 +473,68 @@ <h1>Search input is not valid</h1>
       });
     })
 
+    function calculate_new_url(param, $select) {
+      var uri = URI(window.location.href);
```
[the remark this answers is not in the document]
Comment (kindly) [reply grantnav/frontend/templates/search.html:490]: yep.  The library is still nicer to use though and as it was already used I thought I might as well continue using it.

**Review by michaelwood — APPROVED** [review]
Great bit of refactoring

**Pull request merged**